Get last updates and update to gnome 42

fer-moreira opened the pull request.

okseb commented:
Any update on this?

DaveSpe commented:
Can we get an update on this? Gnome 42 is out there in the wilds.

roryst commented:
gnome 42, and everything breaking (as always), make me switch to cinnamon. i recommend it

ran-dall commented:
For those interested in trying this out, on Gnome 42, you can pull the repo, patch it, and build it...as follows...

1. Pull the patch.
```shell
$ curl -L https://patch-diff.githubusercontent.com/raw/fer-moreira/floating-dock/pull/12.patch > ~/Desktop/gnome42.patch
```

2. Pull the repo. (I pull it into `~/.local/share/gnome-shell/extensions/floating-dock`)
```shell
$ git clone fer-moreira/floating-dock floating-dock
```
3. Change into `floating-dock` directory.
```shell
cd ./floating-dock
```

3. Apply patch
```shell
$ git apply --check --ignore-space-change --ignore-whitespace --3way  '~/Desktop/gnome42.patch'
```

NOTE: The patch expects `schemas/org.gnome.shell.extensions.floating-dock.gschema.xml` to be named `schemas/org.gnome.shell.extensions.dash-to-dock.gschema.xml`; as does the `Makefile` you'll need to use.

4. Make
```shell
$ SASS=dart make
```

5, Install
```shell
$ make install
```

---

It should be noted there are still some errors; for example, on GNOME Extensions, it will kinda break, displaying a `Main.panel._leftCorner is undefined` error. It also shows that it can't be activated on GNOME Extensions, but clearly, it works.

![Screenshot from 2022-05-29 06-34-30](https://user-images.githubusercontent.com/17261190/170871634-dd01895f-ff2e-4987-b8b3-0babd20efee5.png)
![Screenshot from 2022-05-29 06-36-06](https://user-images.githubusercontent.com/17261190/170871678-58c2c340-0371-45d1-8f52-ce355898ed24.png)

Another minor error (a warning) will be displayed when performing Step 4.

```shell
$  SASS=dart make
glib-compile-schemas ./schemas/
sass --no-source-map _stylesheet.scss stylesheet.css
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($base-padding, 4) or calc($base-padding / 4)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
13 │ $dash_spacing: round($base_padding / 4);
   │                      ^^^^^^^^^^^^^^^^^
   ╵
    _stylesheet.scss 13:22  root stylesheet

Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($dash-bottom-margin, 4) or calc($dash-bottom-margin / 4)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
19 │ $dock_side_margin: $dash_bottom_margin / 4;
   │                    ^^^^^^^^^^^^^^^^^^^^^^^
   ╵
    _stylesheet.scss 19:20  root stylesheet

Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($val, 4) or calc($val / 4)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
25 │     @return round($val / 4);
   │                   ^^^^^^^^
   ╵
    _stylesheet.scss 25:19  shrink()
    _stylesheet.scss 89:38  root stylesheet

Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($dash-border-radius, 2) or calc($dash-border-radius / 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
91 │                     border-radius: $dash_border_radius / 2;
   │                                    ^^^^^^^^^^^^^^^^^^^^^^^
   ╵
    _stylesheet.scss 91:36  root stylesheet

Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($dash-padding, 2) or calc($dash-padding / 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
63 │                     margin: ($dash_spacing + ($dash_padding / 2)) 0;
   │                                               ^^^^^^^^^^^^^^^^^
   ╵
    _stylesheet.scss 63:47  root stylesheet

msgfmt -c po/ar.po -o po/ar.mo
msgfmt -c po/cs.po -o po/cs.mo
msgfmt -c po/de.po -o po/de.mo
msgfmt -c po/el.po -o po/el.mo
msgfmt -c po/es.po -o po/es.mo
msgfmt -c po/eu.po -o po/eu.mo
msgfmt -c po/fr.po -o po/fr.mo
msgfmt -c po/gl.po -o po/gl.mo
msgfmt -c po/hu.po -o po/hu.mo
msgfmt -c po/id.po -o po/id.mo
msgfmt -c po/it.po -o po/it.mo
msgfmt -c po/ja.po -o po/ja.mo
msgfmt -c po/nb.po -o po/nb.mo
msgfmt -c po/nl.po -o po/nl.mo
msgfmt -c po/pl.po -o po/pl.mo
msgfmt -c po/pt_BR.po -o po/pt_BR.mo
msgfmt -c po/pt.po -o po/pt.mo
msgfmt -c po/ru.po -o po/ru.mo
msgfmt -c po/sk.po -o po/sk.mo
msgfmt -c po/sr@latin.po -o po/sr@latin.mo
msgfmt -c po/sr.po -o po/sr.mo
msgfmt -c po/sv.po -o po/sv.mo
msgfmt -c po/tr.po -o po/tr.mo
msgfmt -c po/uk_UA.po -o po/uk_UA.mo
msgfmt -c po/zh_CN.po -o po/zh_CN.mo
msgfmt -c po/zh_TW.po -o po/zh_TW.mo
```

CristhoferMF commented:
it will be merged?
appears me same error
![image](https://user-images.githubusercontent.com/36576423/175399907-b643b4f9-46d8-4075-9dd2-c49885a08425.png)

martinszeltins commented:
What is the status of this? Since Ubuntu 22.04 is out, it would be nice to upgrade this extension to support GNOME 42

okseb commented:
Any news on this?

lookto commented:
A little update would be much appriciated :)

urmatej committed:
Created sl-po

Created sl.po with Slovenian translation

stuarthayhurst committed:
appIcons: Use Mtk.Rectangle instead of Meta.Rectangle

Meta.Rectangle has been deprecated, fixes logs being flooded with
'Meta.Rectangle is deprecated, use Mtk.Rectangle instead'

xalt7x committed:
Fix dashtodock.pot generation

Some strings are not generated due to redundant option "-k"
(leftover from commit 51db80abc3ec5c6c0621a0ba74f04919cb3f0b12)

In result most translations lack:
msgid "Show Details"
msgid "Launch using Integrated Graphics Card"
msgid "Launch using Discrete Graphics Card"
msgid "Failed to launch “%s”"

urbalazs committed:
Update Hungarian translation

xalt7x committed:
Update Ukrainian translation

3v1n0 committed:
metadata: Bump to version 88

3v1n0 committed:
docking: Do not manually dispose the settings object

This is handled by the garbage collector, so no need to handle this
ourself.

3v1n0 committed:
docking: Use St.IconTheme directly, without wrappers

The old code was using Gtk as fallback to support old versions but this
is not the case anymore.

3v1n0 committed:
dependencies/gi: Do not import Gdk or Gtk into the shell

None is using them, so let's drop the dependencies

3v1n0 committed:
Bump version to v89

3v1n0 committed:
appIcons: Add support for translating plural forms

Closes: #2100

hasecilu committed:
Update Spanish translations

xalt7x committed:
Add missing third argument

xalt7x committed:
Use the function ngettext directly

With this change the plural strings are generated in the potfile

xalt7x committed:
Update Ukrainian translation (sync with PR #2102)

proninyaroslav committed:
Change "Show Details" to GNOME's "App Details"

proninyaroslav committed:
Fix auto format indents

Amr-Ibra committed:
Update German translation

smedir committed:
Remove any dependency on dockmanager in desktopIconsIntegration.js

Dockmanager can be null, and this spams the log if desktopIconsIntegration class
is initialized and used.

desktopIconsIntegration class is initialized and called from the extension and
really should not have any other dependencies other than the shell,
it is meant to be used and can be used as is by any extension if called as is
independently without modifications.

Fix resource path for Extension

pwiecz committed:
Fix Polish translation of "Empty Trash"

My understanding is that "Empty" here is a verb in imperative, not an adjective.
The previous translation was correct only if "Empty" was as adjective.

vanvugt committed:
Merge pull request #2140 from pwiecz/master

Fix Polish translation of "Empty Trash"

vanvugt committed:
Merge pull request #2118 from Amr-Ibra/german-translation

Update German translation

vanvugt committed:
Merge pull request #2107 from xalt7x/ukrainian-translation

Update Ukrainian translation (sync with PR #2102)

taoky committed:
intellihide: enable/disable unredirect when dock shows/hides

sergio-costas committed:
Update for Gnome Shell 46

add_actor() has been removed after being deprecated, and it must
be replaced with add_child(). Also, there are two add methods
that also have to be changed.

sniirful committed:
Fix for blurry background when opening a new application

sniirful committed:
Simplify animation duration expression into one-liner

Co-authored-by: Marco Trevisan <mail@3v1n0.net>

sergio-costas committed:
Right detection of barriers

sergio-costas committed:
Fix Overview mode

There is a property, DockManager.spacing, that isn't defined
anywhere, but is used when calling two methods inside
(maybeAdjustBoxToDock and maybeAdjustBoxSize). The result is
that these methods are being called with an undefined variable,
and they use its value to calculate the position of a box, thus
resulting in invalid allocation values.

This patch fixes this by ensuring that any invalid value in
the parameter will be converted to zero.

Another possibility would be to fully remove the parameter,
since it seems neither to be set the parameter anyhwere, nor to
be called those functions from any other place.

sergio-costas committed:
Code simplification

sergio-costas committed:
Fix position error

This patch fixes the placement error in the app icons

sergio-costas committed:
Remove the gray background in each icon

sergio-costas committed:
Show the currently focused application

sergio-costas committed:
Fix hover highlight in Gnome Shell 45

Also remove addActor(), because it was used in a single element.

sergio-costas committed:
Nitpicks requested

sergio-costas committed:
appIndicators: Update progress bar colors to be more modern

As commented in https://github.com/ubuntu/yaru/issues/4016, the
progress bar, with its 3D gradient track, doesn't fit very well
in the current Gnome/Yaru theme.

This patch changes this, moving the track to a planar look, and
also allows to set the colors in the CSS.

sergio-costas committed:
appIndicators: Use track instead of background

sergio-costas committed:
stylesheet: Allow to use CSS to fully set gradient colors

vanvugt committed:
windowPreview: Use set_child instead of add_child

While it's safe to use either in GNOME 46, it is not always safe to use
(clutter_actor_)add_child directly in GNOME 45. Because the notification
StBin requires to update its internal state wasn't added until
gnome-shell@ed3b8d093e introduced in 46.alpha.

At least (st_bin_)set_child remains the same in all GNOME versions, and
is what the old (clutter_container_)add_actor used to call in GNOME 45.

There may be other places where we need to change add_child to set_child
for backward compatibility with StBin, but if they exist then they're not
causing problems yet.

Closes: https://bugs.launchpad.net/bugs/2056506
Fixes: 4945dbf592a9f5641416588cd7ae23b7a357a1ec

sergio-costas committed:
Remove unneeded settings in overview-til

taoky committed:
docking: fix window picker space for multi-monitor environments

sergio-costas committed:
Fix odd rectangle in apps view

vanvugt committed:
Merge pull request #2146 from sergio-costas/update_to_gnome_shell_46

Update for Gnome Shell 46

vanvugt committed:
docking: Hard code the startup animation time

Because gnome-shell stopped exporting the value in 45.0 it was being
treated as undefined, and therefore we had no startup animation.

vanvugt committed:
docking: Use the upstream gnome-shell startup animation code

Rather than trying to duplicate it. And when the upstream animation is
done we can run our own slide-in animation.

Closes: https://bugs.launchpad.net/bugs/2058468
Closes: https://github.com/micheleg/dash-to-dock/issues/2110

taoky committed:
docking: adapt ControlsManagerLayout spacing for GNOME 46

sergio-costas committed:
Fix dot height in default mode (#2168)

Fix https://github.com/micheleg/dash-to-dock/issues/2160

jichizhang committed:
intellihide: treat fullscreen windows as maximized

K1ngjulien committed:
fix deprecated "St.ScrollView.[hv]scroll" property

only used to access adjustments, which can be directly accessed now.

see: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3020

yakushabb committed:
i18n: Fix sl-po file extension

3v1n0 committed:
ci/make: Bump to checkout@v4

3v1n0 committed:
make: Continue using flat config

3v1n0 committed:
locations: Remove unused variable

taoky committed:
intellihide: Fix unexpected unredirection disable with overview

When overview shows, this._animateIn is called in
_onOverviewShowing(). And when overview hides, _updateDashVisibility()
might call this._animateIn() again, which makes the unredirection always
disabled and brings performance issue in some cases.

This commit replaces the counter for unredirection by a boolean value,
to make logic more clear, and disabling/enabling unredirection balanced.

3v1n0 committed:
appIcon: Allow customizing the showAppsIcon based on the session mode

3v1n0 committed:
Release dash-to-dock v90

3v1n0 committed:
docking: Use the signal handler to disconnect on startup-completed signal

As requested by the g.e.o review, it's not really a big deal though
since this signal is going to really last few frames and the extension
is not really unloaded by that time.

3v1n0 committed:
locations: Do not cancel operations after a volume has been umounted / removed

In case we're unmounting a volume we were optionally removing its icon
but this was also leading cancelling the operation that in the Gio queue
caused the action not to be fully performed.

So once, we've removed a volume or umounted a mount, we don't really
care about cancelling the ongoing operations since they should be done
already.

LP: #1993054

3v1n0 committed:
locations: Do not warn if a file info has not the icon attribute

We were still getting a null icon but Gio was warning about so we
can just check if it has such attribute first.

3v1n0 committed:
Release dash-to-dock v91

bc-lee committed:
i18n: Add Korean translation

sergio-costas committed:
Fix window preview in Gnome 45

When setting the middle button click as "show window preview",
the previews aren't shown. Instead, an empty rectangle appears
and a lot of errors are shown in the log.

This patch fixes this.

Fix #2202

sergio-costas committed:
Move .actor test to its own function

3v1n0 committed:
Releasing dash-to-dock v92

3v1n0 committed:
locations: Ask again for password on encryption failures

This sadly requires parsing the error strings since there are not good
APIs yet, but that's how upstream handles this too, so let's accept it
for now.

See: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3300

sergio-costas committed:
Fix communication with DING

The extension state naming has changed from gnome shell 45 to
gnome shell 46, so the code to notify margins to DING wasn't
being able to detect when an extension was active, and so it
didn't prevent to put icons below the dock.

This patch fixes it.

sergio-costas committed:
Apply changes in all cases

sergio-costas committed:
Fix style

vanvugt committed:
docking: Fix allocation failure in GNOME 47 that completely broke the shell

Since this._container no longer exists (gnome-shell@b58119d5c6?) in
GNOME 47, allocating the entire overview would fail and the shell was
non-functional.

Fixes:
```
Gjs-CRITICAL **: 17:10:01.720: JS ERROR: TypeError: actor is undefined
findIndexForActor@resource:///org/gnome/shell/ui/layout.js:992:22
findMonitorForActor@resource:///org/gnome/shell/ui/layout.js:999:26
_prepareMainDash/<@file:///home/dan/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:2233:52
@resource:///org/gnome/shell/ui/init.js:21:20
```

sergio-costas committed:
Don't show error messages when disabling

Since the destroy() function can be called several times, it is
paramount to don't call objects that have been freed in
previous calls.

sergio-costas committed:
UDENG-3111: allow to disable an icon in the dock during updates (#2240)

* UDENG-3111: allow to disable an icon in the dock during updates

For the new RAA we need to be able to mark an icon as "disabled"
while it is being updated. The current Unity LauncherAPI
(https://wiki.ubuntu.com/Unity/LauncherAPI) doesn't support
this, so this patch adds an extra option for this.

https://docs.google.com/document/d/1--DgBRl6AqNiyjW_luOjl1dDzsIZPlF0Ukc7INW9_XQ

Voltara committed:
Fix window selected from preview not getting focus

Fixes #1972

vanvugt committed:
metadata.json: Enable support for gnome-shell 47 (#2248)

Closes: https://github.com/micheleg/dash-to-dock/issues/2246

3v1n0 committed:
appIcons: Make possible to override updating icons style from css

We still use a fallback opacity if the theme doesn't provide a value,
but otherwise we check on theme value.

3v1n0 committed:
appIcons: Do not allow launching a new window if the app is updating

3v1n0 committed:
appIcons: Hide launch menu items and generic actions if the app is updating

3v1n0 committed:
Releasing dash-to-dock v93

3v1n0 committed:
notificationsMonitor: Try to get appId from the source if set

3v1n0 committed:
appIcons: Sync comment change with ubuntu-dock

3v1n0 committed:
appIcons: Fix comment typo

3v1n0 committed:
appIcons: Show an OSD error if trying to launch an updating app

3v1n0 committed:
cleanup: Fix various comments and code typos

3v1n0 committed:
appIcons: Remove duplicate source definition on DockAppIconMenu

Avoid keeping track of source twice, and cleanup the usage of it
when we only rely on the related app.

3v1n0 committed:
stylesheet: Add missing declaration for overview tile states

This changed as part of upstream commit:
 https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/745667a825

3v1n0 committed:
appIcons: Allow to update the dot style offset in x too

3v1n0 committed:
appIcons: Inform in menu items that the app is being being updated

3v1n0 committed:
appIcons: Always add a disabled menu item with the app title

3v1n0 committed:
appIcons: Delete menu instance from app icon on destruction

This doesn't happen upstream, so let's handle it ourselves

3v1n0 committed:
appIcons: use upstream definition of shouldShowTooltip

3v1n0 committed:
docking: Patch AppIcon to have more consistent Pin labels

Use the very same wording for both the dock icons and the overview
icons so that both match.

This implies breaking the translations but we're fine at this point.

Probably better not to use the old "favorites" naming yet.

3v1n0 committed:
appIcons: Use a labeled menu item separator for application title

It's more integrated and uses less space

3v1n0 committed:
appIcons: Drop code that is not used anymore by any supported version

3v1n0 committed:
appIconIndicator: Move style handling for updating state here

This is the UI class, while the appIcons one is only supposed to be a
model class

3v1n0 committed:
appIcons: Setup unity indicator properties on construction

These properties may have been set already by any application around
(hey snapd!), so we need to initialize the values properly and not only
rely on updates.

So that when application is launched we pick those values

3v1n0 committed:
appIconsDecorator: Support decorating app icons with emblems everywhere

If unity indicators are enabled we should show them allover the places,
but we didn't so far, so let's add support for patching all the kinds of
icons so that we can define the behavior when this happens

3v1n0 committed:
launcherAPI: Avoid using regexes to replace all values in a string

3v1n0 committed:
utils: Support injecting properties that are not on the main object

3v1n0 committed:
appIndicators: Allow theming progress offsets

And use it to move the progressbar next to the middle of an icon if
it includes a label

3v1n0 committed:
appIconsDecorator: Update icons style on updating changes

3v1n0 committed:
appIconsDecorator: Do not allow launching any application in updating state

3v1n0 committed:
appIconsDecorator: Hide all menu items except unpinning ones on updating

Do not allow performing any action on updating apps

3v1n0 committed:
Release dash-to-dock v94

3v1n0 committed:
appIconsDecorator: Use the generic icons signal handler for result

3v1n0 committed:
appIconsDecorator: Destroy indicators on removal

We were removing the container set, but not its children, so let's
cleanup everything when indicators are cleared.

Closes: #2272, #2270

3v1n0 committed:
appIconsDecorator: Drop unneeded eslint ignores

3v1n0 committed:
appIconsDecorator: Use a signal signals handler with labels

No need to use two objects, we can use labeled connections to simplify
our case

3v1n0 committed:
appIconIndicators: Use lambda-style callbacks for better debugging

3v1n0 committed:
utils: Remove signal handler item on object destruction

If we're tracking an object destruction and that object gets destroyed,
we need to remove the signals handler storage item, or we'd end up
trying to disconnect from it again when destroying the signals handler.

Closes: #2270

3v1n0 committed:
Release dash-to-dock v95

proninyaroslav committed:
Add `Dot` running indicator style

sergio-costas committed:
Fix "Show Applications" button

The "Show Applications" button fails because the
Overview.ANIMATION_TIME property is now read-only (it has a
"const" prefix), so it's not possible to overwrite it to show
the applications list "instantaneously".

This patch removes the option of disabling the animation, thus
fixing the problem.

3v1n0 committed:
utils: Add clamp function and use it for Indicators values

3v1n0 committed:
appIndicators: Add function to read theme values in a more JS way

Let's handle unset values as undefined, as we should do

3v1n0 committed:
appIconIndicators: Make progress bar paddings styled

Don't use normal padding property here because we want the padding
to be proportional to the icons sizes

3v1n0 committed:
stylesheet: Make alignment in progress overview icons more consistent

3v1n0 committed:
appIconIndicators: Make progress bar height factor styled

3v1n0 committed:
stylesheet: Use lighter progress bar on bigger overview icons

That includes the search results and the app folders contents

proninyaroslav committed:
Update Russian translations

sergio-costas committed:
Merge pull request #2277 from proninyaroslav/russian-translations

Update Russian translations

3v1n0 committed:
appIconIndicators: Do not add notification badge code unless needed

Only create the label and items if we are showing stuff, otherwise it's
just a waste.

Also update it on scale factor changes

3v1n0 committed:
appIcons[Decorator]: Improve wording on updating messages

3v1n0 committed:
Release dash-to-dock v96

Amr-Ibra committed:
Update German translation

vanvugt committed:
theming: Add support for Cogl.Color since Clutter.Color is deprecated in 47

Closes: https://github.com/micheleg/dash-to-dock/issues/2284

3v1n0 committed:
appIcons: Show app details menu for snap applications

If an application is a snap package, and the snap store is installed
we can show the snap packages details as we do for other non-snap
applications if GNOME Software is installed

3v1n0 committed:
appIconsDecorator: Keep showing the details items when updating

3v1n0 committed:
theming, appIconIndicators: Support both Cogl.Color and Clutter.Color

vinchatl committed:
Update French translation

3v1n0 committed:
Release dash-to-dock v97

3v1n0 committed:
theming: Give Clutter.Color priority over Cogl.Color

While now there's only Cogl.Color, in previous versions we had both so
giving Cogl priority broke the theming support

Closes: #2307

3v1n0 committed:
Release dash-to-dock v98

3v1n0 committed:
utils: Use an extra parameter for property injector constructor

By adding parameters in commit 611690cc33 we ended up hiding the base
object constructor parameter, and so the support for automatic
disconnection on parent object destruction.

Fixes: #2297

3v1n0 committed:
Release dash-to-dock v99

sergio-costas committed:
Fix dot position on Gnome 46.2 and greater (#2315)

This patch fixes the dot in Default mode when not enabling "Few
customizations".

Tested in Ubuntu Noble, Fedora 46 (100% and 200% scale), and
Ubuntu Oracular (both in Ubuntu Desktop and Classic Gnome
sessions).

Closes: #2222

xalt7x committed:
Make the 'Pin to Dock' label translatable

With the single underscore (_) we can re-use original translation.
However, for the 'Pin to Dock' there's no match since GNOME 42-47 original label is 'Pin to Dash'.
Using two underscores (__) as a translation function resolves issue 2330

xcpn commented:
Very unfortunate this project is dead.

Been waiting for updates to come for GNOME 43+ so I can upgrade to Debian 12, but it seems like it'll never happen.

Is there any guide or a plan for someone to fork it so we can use it? I really dislike dash-to-dock and floating-dock was perfect.

sergio-costas committed:
Merge pull request #2331 from xalt7x/fix-pin-to-dock-string

Make the 'Pin to Dock' label translatable

xalt7x committed:
Update uk_UA.po

vanvugt committed:
metadata.json: Enable GNOME 48 support

vanvugt committed:
docking: Generalize unredirect toggling to support GNOME 48 and earlier

Closes: https://github.com/micheleg/dash-to-dock/issues/2347

sergio-costas committed:
CSS: fix syntax style

The SASS documentation specifies that a list of elements for
the @each statement should be defined without square brackets,
but there are two definitions in the .scss file that uses them.
This patch fixes it.

source: https://sass-lang.com/documentation/at-rules/control/each/

cesar19004 committed:
fix autohide after drag event

3v1n0 committed:
appIcons: Be consistent on menu items styles

So for the dash-to-dock menu item use a popup separator as well with the
dash to dock name and a settings menu item.

3v1n0 committed:
Release dash-to-dock v100

aleasto committed:
dash: Support new API for retrieving scroll adjustment

The previous API is removed in GNOME 48.

LP: #2102002

aleasto committed:
dash: Simplify scroll computation by always using the SMOOTH events

Clutter will always emit a smooth scroll event, marked as emulated if it was
originated by a mouse that only supports fixed-step scrolling.

Instead of ignoring emulated events to avoid handling the same event twice,
let's switch to only handling the smooth events.

This is functionally equivalent with the native backend, and fixes
double-scrolling when running in the nested backend, where events are never
marked as emulated.

aleasto committed:
dash: Handle vertical scrolling on a horizontal dock

Even if the dock is horizontal, we should handle vertical scroll events
as emitted by mouse devices.

We probably do not want the vice-versa on a vertical dock instead.

Closes: #2301

bllyds999 committed:
add 'Pin to Dock' zh_CN

Tommimon committed:
Fixing "Pin to Dock" translation in app grid menu

MorganAntonsson committed:
Update Swedish translation

bittner committed:
Correct German translation of threshold

3v1n0 committed:
Release dash-to-dock v101

3v1n0 committed:
docking: Drop unused variable definition

3v1n0 committed:
ci: Lock eslint version to 9 for now

We are targetting debian/ubuntu where eslint is quite old, so we don't
want to switch configuration format yet

3v1n0 committed:
metadata: Add support for GNOME 49

It works properly for now, not changing it upstream yet though, as it
may still be a temporary case

Amr-Ibra committed:
locations: Fix action typo leading to error not being handled

Amr-Ibra committed:
locations: Fix unmount typo

Amr-Ibra committed:
Update German translation

x9a committed:
po: update Arabic translation

Commit:
Settings UI update for pressure_threshold_adjustment fields: upper, step-increment and page-increment

tigertall committed:
Update zh_CN.po, 修正笔误

3v1n0 committed:
appSpread: Make the search entry insensitive during app spread

Otherwise we could still type on the search entry and focus it

3v1n0 committed:
windowPreview: Fix typo

3v1n0 committed:
appIcons: Do not call missing functions in GNOME 49

These have been removed as part of clutter refactor

3v1n0 committed:
appIcons: Do not throw if parent class does not define vfunc

3v1n0 committed:
appIcons: Use clutter gestures if defined to show popup

We need to re-implement this as it does not seem possible to re-use
upstream code here

3v1n0 committed:
appIcons: Disable the show-apps menu in ubuntu-dock

We can avoid having this delta downstream, so let's just use the same
code upstream as this would work in both cases

3v1n0 committed:
locations: Avoid using hardcoded strings for removable actions

Use instead an enum-like definition

3v1n0 committed:
locations: Use switch cases to handle removable actions

It makes things easier to read

3v1n0 committed:
po: Update translations to be adjusted for the typo

3v1n0 committed:
metadata: Release dash-to-dock v102

3v1n0 committed:
locations: Use DesktopAppinfo from GioUnix

This is has always been the case, although we had a wrapper for Gio too.

But this is not anymore the case in girepository-2.0 based gjs, so let's
just use the proper namespace everywhere instead of warn

PixelcrafterEXE committed:
Fix Autohide Location calculation bug (Issue #2413)

lasaczka committed:
Add Belarusian translation

Benjamin-Loison committed:
Use keyboard formatting in `README.md`

Personal notes: [Benjamin_Loison/dash-to-dock/issues/12](https://codeberg.org/Benjamin_Loison/dash-to-dock/issues/12)

yakushabb committed:
Update Turkish translation

vanvugt committed:
metadata.json: Declare support for GNOME 50

50.alpha is coming soon and it seems to work well enough to declare support.
This has to be done before Ubuntu gets GNOME 50.alpha.

Commit:
po: add Bulgarian translation

github-actions[bot] committed:
fix: correct French translation typo in Unity API description

Change "utilsée" to "utilisée" in the French translation.

Fixes #2480

3v1n0 committed:
appIcons: Do not handle removal of PopupManager.ignoreRelease()

3v1n0 committed:
docking: Correctly handle add/removing a dock

As when changing settings

3v1n0 committed:
dash, appIcons: Ensure Dash const values are defined

They are not all exported in all the shell versions, so let's define
them if they are not set

3v1n0 committed:
metadata: Release dash-to-dock v103

crayxt committed:
Add Kazakh translation

madmoh committed:
stylesheet: fix show-apps gap to match app icon spacing

vadimmaksimushkin committed:
Fix overview state stuck at WINDOW_PICKER when disableOverviewOnStartup is enabled

EkaterinePapava committed:
po: Add Georgian translation

Commit:
added the missing </kbd> tag for the shell reload key sequence under the "Building" section

aleasto committed:
appSpread: Fix override of mouse input in Panel.ActivitiesButton

In GNOME Shell 50, Panel.ActivitiesButton no longer overrides vfunc_event,
which means that we cannot replace it. Use a Clutter.ClickGesture like
upstream; this will have event priority as it was added last.

aleasto committed:
appSpread: Fix override of keyboard input on Panel.ActivitiesButton

Since GNOME Shell 45, the key_release_event vfunc takes a non-subtyped
Clutter.Event argument.

aleasto committed:
Revert "Fix Autohide Location calculation bug (Issue #2413)"

Clutter.Actor.get_transformed_position() accounts for animations etc...
which means that it cannot be used to compute the "theoretical" global position
of the dock, which indeed might be part of an animation at the moment that the
method is called.

This reverts commit 3e0c24f7b09708757e70565a6e449b6143cd2b08.

Closes: https://github.com/micheleg/dash-to-dock/issues/2501

mimimiku778 committed:
intellihide: Fix dock not hiding when windows touch dock edge

The overlap test used strict greater-than (>) to compare window edges
against the dock target box.  On a bottom dock at y=1080 with a
maximized window whose bottom edge is exactly at y=1080, the test
`1080 > 1080` evaluates to false, incorrectly reporting no overlap.
This causes the dock to show and stay visible because intellihide
keeps finding overlap=FALSE on every recheck.

Change > to >= so that windows touching the dock edge are considered
overlapping, which is the correct behavior for intellihide — a window
filling the screen up to the dock should trigger the dock to hide.

Also skip the immediate _doCheckOverlap() in _windowCreated since the
new window's frame_rect is still 0x0 at that point, which can cause a
false "no overlap" flash.  The notify::allocation signal connected by
_addWindowSignals already triggers a recheck once the window has a
real geometry.

EriksRemess committed:
Initial Latvian translation

Armatura98 committed:
Fix typo in OverlapStatus assignment: this.status -> this._status

Daniel-Escamilla committed:
docking: drop direction argument from workspaceSwitcherPopup.display()

GNOME Shell dropped the direction parameter from
WorkspaceSwitcherPopup.display() in commit [1] included starting from
GNOME shell 42.

Ubuntu-dock still passes direction as the first argument, which lands in activeWorkspaceIndex. Since direction is never a valid workspace index, no workspace dot is ever highlighted as active.

LP: #1966681

[1] https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/209d8c7f10b2753

Rajat-XR committed:
po: Add Romanian translation

3v1n0 committed:
build: Use POTFILES.in to define translatable files

This is more in line with what we do in most projects and would make a
future meson switch easier

3v1n0 committed:
ci: Generate potfile

3v1n0 committed:
dash: Track icon's signals using connectObject

Avoids accessing to destroyed target objects

3v1n0 committed:
docking: Adjust syntax to meet eslint rules

3v1n0 committed:
metadata: Release dash-to-dock v104

GabrielAyala committed:
po: improve pt_BR translations

AsciiWolf committed:
Do not mark the whole Kazakh translation as fuzzy

AsciiWolf committed:
Update Czech translation.

DawidNet7 committed:
Add Polish translations for 'Unpin' and 'Pin to Dock'

vanvugt committed:
docking: Avoid inserting ("tracking") docks during the login animation

Up until GNOME 49 we could be confident that extensions were activated
before the login animation, but that's not true in GNOME 50
(gnome-shell#8917). In GNOME 50 we get constructed _during_ the GNOME
login animation so inserting docks at that moment creates a visible
glitch.

Now we wait until the GNOME login animation is done before starting our
own dock login animation. Or if we're not starting up then insert docks
immediately.

Closes: https://bugs.launchpad.net/bugs/2146516

3v1n0 committed:
docking: Unfocus currently focused window when getting the a11y focus

When the dock gets focused through the Ctrl+Alt+Tab switcher we are
getting the focus, but the currently focused application still preserves
the focused appearance.

Adjust this by ensuring the focus is unset.

See: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4164

LP: #2147922

3v1n0 committed:
docking: Cleanup accessibility focus callback handling

3v1n0 committed:
docking: Correctly check for remote model need

We were always returning true, ignoring any setting.

Closes: #2549

3v1n0 committed:
appIconsDecorator: Keep track of icons and results appIcons separately

We used the indicators set to track all the indicators kinds, however
this was problematic when the 'view-loaded' signal was emitted, because
it lead all the indicators being cleared, and not only the icons ones.

Thus when the destroyed signal was called for an indicator that we
cleared from the set, its source was invalid.

Split the handling of those in two different sets, so that we do not
clear what should not be cleared when signals or events happen

Closes: #2482
LP: #2146632

3v1n0 committed:
appIcons[Decorator]: Adjust to GNOME 49 API to show OSD

Upstream signature of method to show OSD changed upstream during GNOME
49 cycle [1], but it went unnoticed

So adjust it.

[1] https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3751

LP: #2148313

3v1n0 committed:
appIconsDecorator: Track updating icons externally

If an icon update value changes while the dock is destroyed we may not
adjust the updating state.

Also we are adding our properties to an icon that we do not own

3v1n0 committed:
docking: Do not connect multiple times to the same global signal

We were connecting (and never disconnecting) from this settings change,
while we were already monitoring the key in the constructor.

What we did not do, was handling the disable case though, that is now
done.

3v1n0 committed:
docking: Simplify the creation and destruction of the docks

We were not tracking the dock destruction events so we may have
potentially accessed to destroyed dock's when iterating over them

Helps with: lp:2148339

3v1n0 committed:
docking: Disconnect from signals on docks destruction

Automatically disconnect from global signal on docks destruction, and
when the workspace isolation gets disabled or destroyed.

LP: #2148339

3v1n0 committed:
docking: Remove workspace switcher pop-up only if matching

3v1n0 committed:
docking: Disconnect from remote model settings changes

3v1n0 committed:
docking: Disconnect from timeline new-frame signal on destroy

3v1n0 committed:
docking: Remove number overlay timeout from properties

3v1n0 committed:
intellihide: Also disconnect from window destroy signal

We may want to ignore it once intellihide is being disabled

3v1n0 committed:
menuClient: Disconnect from previous menu client signal

3v1n0 committed:
appIconsDecorator: Factorize the appIcon updating notification code

3v1n0 committed:
theming: Only update the theme actor style when the actor is mapped

If the actor is not visible we do not care about updating its style,
also this was causing problems since commit 8aa70efb6, as now the dock
can be tracked (e.g. staged later), thus we do not want to set theming
data while the widget it's unmapped.

3v1n0 committed:
theming: Ensure the overview style is set on startup

3v1n0 committed:
docking: Move back dock tracking to the DockedDash

This adjusts commit 8aa70efb, ensuring that also secondary monitors
docks are tracked, while we were potentially not tracking them anymore
in case the startup animation was not already ran

3v1n0 committed:
docking: Reset position as part of dock tracking

It's something we always do together, so there's no point of duplicating
it in multiple places

3v1n0 committed:
docking: Bind dock position only after initialization

This is something that should only happen only after the dock is staged
so let's ensure this is the case

3v1n0 committed:
docking: Only initialize the dock after the startup has completed

We may otherwise end up doing operations that requires a staged actor
before than we can

3v1n0 committed:
docking: Move the docked dash definition earlier than signal connections

3v1n0 committed:
docking: Iterate over a docks snapshot on destruction

Each dock destroy signal handler removes the dock from the allDocks
array, but doing this while looping on it may lead to skipping elements
and it's not safe.

So loop over a snapshot instead.

Fixes: 13ca2018

3v1n0 committed:
docking: Remove deprecated code for older gjs

3v1n0 committed:
docking: Do not update the custom theme twice

We do this already since 287f2f65d in both the startup and non-startup
cases, so no need to duplicate the effort.

3v1n0 committed:
dash: Skip a workaround in newer shell versions

This has been fixed in mutter upstream now, so we do not need to
workaround it tracking or triggering unneeded events.

3v1n0 committed:
dash: Expose the max-width/height values as properties

This will make possible to use them in bindings

3v1n0 committed:
docking: Use bindings to sync the dock/dash sizes

Ensures things happens automatically with less JS involvement

Helps with lp:2035304

3v1n0 committed:
docking: Only show the dock once fully initialized

Workarounds dock resizing issues during init

LP: #2035304

3v1n0 committed:
docking: Remove dead code since GNOME Shell 42

3v1n0 committed:
metadata: Release dash-to-dock v105

3v1n0 committed:
appIcons: Use the signals handler to track source destruction

So we are disconnecting from it automatically if needed

3v1n0 committed:
utils: Ensure that destroy arguments are passed to the callback

We may potentially use the object value, so let's pass it through

3v1n0 committed:
desktopIconsIntegration: Use the current extension UUID

Avoid looking-up what we already know

3v1n0 committed:
docking: Unset the static box on destruction

Makes the GC happier

3v1n0 committed:
appIconIndicators: Untrack the destroyed area

3v1n0 committed:
docking: Remove scroll timeout on destruction

3v1n0 committed:
docking: Disconnect from pressure barrier signals on destruction

3v1n0 committed:
docking: Only track the workspace popup we manage

Disconnect from workspace popup signals and drop it on destruction

3v1n0 committed:
utils: Disconnect from all signals if destroy is tracked

If we track an object destroy signal, then we can just disconnect from
all the object signals when the destroy events come

3v1n0 committed:
Makefile: Do not ship settings schema on zip file

This is not needed anymore and even wrong

3v1n0 committed:
docking: Use default showAppsButton behavior when used in Ctrl+Alt+Tab

Ctrl+Alt+Tab switcher is used to switch focus to shell elements, and
with dash to dock is currently broken when it comes to select windows
or apps overview states.

So track when this happens and rely on the default behavior

LP: #2148513

sergio-costas committed:
appIcons: Fix DEFAULT dot position

Until now, the dot vertical position was -12, inherited from
code from Gnome Shell. But with the current layout, that value
is wrong and must be zero to ensure that the dot is placed
outside the icon, below it.

This patch fixes it.

3v1n0 committed:
docking: Use a place-holder dash actor during startup

When starting-up we do not want to show the real dash, but neither
to warn when upstream code tries to use the upstream dash features.

So during that phase, replace the upstream dash with a dummy empty actor
that is still part of the UI group.

LP: #2150429

3v1n0 committed:
docking: Simplify the handling of dummy dash case

In case we do not have a dash upstream we can just use a simpler
injection rather than making it dynamic and to work all the times, since
we already rebuild the dash in case the session mode changes, so being
dynamic at call time does not pay off

3v1n0 committed:
docking: do not restore the old dash more than once

In case we're trying to restore the old dash, but the old dash is
already being used, there's no need to go through the restore logic

Commit:
Updated Basque (eu) translation

Commit:
Fixed Basque translation

Trsnaqe committed:
docking: Re-create remote model on state-changed, not changed

  state-changed fires synchronously before 'changed', so AppIcon's
  state-changed handler rebuilt UnityIndicator while _remoteModel was
  still undefined, throwing and leaving icons stuck. Regression of 2852c99.

vanvugt committed:
Makefile: Remember gschemas.compiled for local installs

Or else the extension won't load at all.

Closes: https://github.com/micheleg/dash-to-dock/issues/2595
Fixes: 1ad2628e32 ("Makefile: Do not ship settings schema on zip file")

vanvugt committed:
Don't install gschemas.compiled for $(INSTALLTYPE)==system

Because $(INSTALLTYPE)==system starts by deleting the target directory
so our copy would fail.

Closes: https://github.com/micheleg/dash-to-dock/issues/2602
Fixes: 2bc44a6add ("Makefile: Remember gschemas.compiled for local installs")

sergio-costas committed:
Explain why "scroll over icon" doesn't work

When the "Fixed icon size" setting in the first tab is enabled,
the "Scroll action" in the third tab does nothing when scrolling
over an application icon, although it still works over the
"Show applications" icon.

This PR adds a notification to inform
the user about this and avoid misunderstandings and bug reports
like https://github.com/micheleg/dash-to-dock/issues/2374 .

3v1n0 committed:
docking: Override overview adjustment value to stay hidden during startup

This helps disabling the startup animation

LP: #2153406
Closes: #2582

vanvugt committed:
metadata: Enable support for GNOME 51

It appears to work with 51.alpha so let's get started.

Commit:
docking: Update static box on slider allocation

Keep the intellihide target box in sync when layout allocates the
centered slider, so windows tiled on the right correctly cause the dock
to dodge.

The slider position can also change through layout-driven
(re-)allocation (e.g. the ActorAlign.CENTER centering), which does
not emit notify::x/y, leaving the intellihide target box stale.

Allocation changes now happens also on x/y changes, so we do not need to
cover it twice

Co-Authored-By: Marco Trevisan (Treviño) <mail@3v1n0.net>

Closes: #2413

3v1n0 committed:
docking: Make the OverviewAdjustment value override settable

While it must be a no-op, we don't want to warn when the shell tries to
write on it

3v1n0 committed:
docking: Ru-use overview controls property

3v1n0 committed:
utils: Improve disconnection of signals when parent objects are destroyed

Avoid retaining signal handlers for objects that were destroyed before
the handler itself.
Monitor connected objects for destruction and remove their tracked
signals without attempting to disconnect them again.

Also clean up the parent's tracked signals while handling its
destruction.

Helps with: LP: #2150670

3v1n0 committed:
utils: Monitor connected objects for destruction

Ensure tracked signal handlers are removed when a connected object is
destroyed, avoiding stale signal references.

Helps with: LP: #2150670

3v1n0 committed:
docking: Stop tracking the old dash when it gets destroyed

Do not keep track of a dead object when this happens

3v1n0 committed:
extension: Disable the extension on shutdown

When the shell is shut down we need to stop all the signals and idle's
that may still run, or we may access to invalid data.

The shell does not do it by default, so let's do it in any shell version
so that we won't cause potential crashes on shutdown

LP: #2150670

3v1n0 committed:
utils: Simplify code to get windows by object path

3v1n0 committed:
metadata: Release dash-to-dock v106

3v1n0 committed:
docking: Avoid accessing to main dock before staging it

If the dock is horizontal we may try to access to the main dock height,
while the dash is not yet in the stage.

So override the get_preferred_height() method twice to ensure that we
never return a valid size but also that we handle it once our dash is
set

LP: #2150429

3v1n0 committed:
docking: Define a dummy showAppsButton on the dummy dash

The code may try to access to it, so better to define a dummy content

LP: #2150429

3v1n0 committed:
docking: Avoid adjusting size during startup

It may access to dock size while unstaged

LP: #2150429

paul-geoghegan committed:
appIconIndicators: Announce unread notification count to screen readers

Unread counts were only shown as a visual badge, so screen reader
users had no way to know an app had pending notifications.

Update the accessible name whenever the count changes, using the
parent DashItemContainer in the dock and the icon/label directly in
the overview, since neither has a single shared actor that both
contexts read from.

Closes #2665
